proxy/request_handler.go

```diff
@@ -242,11 +244,15 @@ func (d *RequestHandler) HandleRequest(r *http.Request, rl *rule.Rule) (session
 
 	if !found {
 		err := errors.WithStack(helper.ErrUnauthorized)
-		d.r.Logger().WithError(err).
-			WithFields(fields).
-			WithField("granted", false).
-			WithField("reason_id", "authentication_handler_no_match").
-			Warn("No authentication handler was responsible for handling the authentication request")
+		// found will only be true if an authenticator was found AND authorisation was granted so check
+		// that this wasn't just a problem with the request
+		if err.Error() != helper.ErrUnauthorized.ErrorField {
```

Comment:
This is always false because we set the error to `err := errors.WithStack(helper.ErrUnauthorized)` right before - so this can be removed :)

Reply:
i see your point :), guess i need to think about this a bit more.

the problem is, if this if statement is removed, then invalid credentials or an expired token will result in `level=warning msg=No authentication handler was responsible for handling the authentication request` getting logged, which is inaccurate and misleading.